File: 6. MUP - vozila (saobracajne i vozacke dozvole)/mup-vozila/src/main/java/ftn/euprava/mupvozila/model/RequestForDrivingLicence.java
package ftn.euprava.mupvozila.model;

import ftn.euprava.mupvozila.model.enums.DrivingLicenceType;
import ftn.euprava.mupvozila.model.enums.RequestStatus;
import lombok.*;

import javax.persistence.*;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@ToString
//Lombok end
@Entity
public class RequestForDrivingLicence {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private DrivingLicenceType drivingLicenceType;

    @Column(nullable = false)
    private String citizenId;

    @Column
    private String employeeId;

    @OneToOne
    @JoinColumn(name = "driving_licence_id")
    private DrivingLicence drivingLicence;

    @Enumerated(EnumType.STRING)
    private RequestStatus requestStatus;

}

File: 6. MUP - vozila (saobracajne i vozacke dozvole)/mup-vozila-frontend/src/app/_models/request-for-driving-licence.ts
import { DrivingLicence } from "./driving-licence";

export class RequestForDrivingLicence {

  id!: number;
  drivingLicenceType!: string;
  citizenId!: string;
  employeeId!: string;
  drivingLicenceDTO!: DrivingLicence;
  requestStatus!: string;

}
